sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -75,57 +82,55 @@ object ScalaNativePluginInternal {
     }
   )
 
-  lazy val scalaNativeConfigSettings: Seq[Setting[_]] = Seq(
-    nativeTarget := interceptBuildException {
-      val cwd   = nativeWorkdir.value.toPath
-      val clang = nativeClang.value.toPath
+  def scalaNativeConfigSettings: Seq[Setting[_]] = Seq(
+    nativeTarget in nativeLink := interceptBuildException {
```

Comment:
I still think we should remove all the `in nativeLink` that were not there before. We can debate whether it would be better to have them, but that is a change in semantics compared to before this PR, and since this PR and only a "refactoring", it definitely should not change semantics.